### DISCUSSION TITLE
Publication used for TautomerEnumerator

### DISCUSSION THREAD

**Discussion by bertiewooster** [rdkit/rdkit#6997] [Q&A] [answered]
"[Tautomer enumeration improvements](https://www.rdkit.org/docs/BackwardsIncompatibleChanges.html#tautomer-enumeration-improvements)" in the Release 2022.03 backwards incompatible changes includes:

> The rules for tautomer enumeration in MolStandardize have been updated to more closely match the rules in the original publication. These changes primarily consist of making the rules more specific; the consequence is that less tautomers will be generated with this version. The previous rules can still be accessed via the function GetV1TautomerEnumerator() (Python) or getV1TautomerEnumerator() (C++)

What "original publication" is that referring to?

P.S. I'm curious about examples where the new TautomerEnumerator returns fewer examples than the previous rules V1.

**Comment by greglandrum** [comment, marked as answer]
> What "original publication" is that referring to?

This is actually in the documentaiton (which is not a given, and you have to know where to look: https://doi.org/10.1007/s10822-010-9346-4)
>The enumeration rules are inspired by the publication: M. Sitzmann et al., “Tautomerism in Large Databases.”, JCAMD 24:521 (2010) https://doi.org/10.1007/s10822-010-9346-4

**Comment by greglandrum** [comment]
> P.S. I'm curious about examples where the new TautomerEnumerator returns fewer examples than the previous rules V1.

That's not something I've noticed. The code change adds a missed case to the enumeration rule set, so at first glance you'd expect it to always produce more tautomers, but I suppose that could still result in a smaller number of tautomers in the end because of how the transformations interact with each other.

**Reply by bertiewooster** [reply]
Oh, interesting. I just took the release note's word for it that the new algorithm produces fewer results. In general I'm curious about cases where the two algorithms produce different sets or numbers of results.

**Reply by bertiewooster** [reply]
I did an [empirical comparison of the two RDKit tautomer enumeration algorithms](https://bertiewooster.github.io/2024/05/01/Tautomer-Sources-Comparison.html) and found the the older one, GetV1TautomerEnumerator, produces about 5 more tautomers per molecule than the updated algorithm TautomerEnumerator. This was on the dataset [Tautomer Structures Extracted from Experimental Literature](https://cactus.nci.nih.gov/download/tautomer/) compiled by Marc Nicklaus and co-workers, the same group that published the paper that RDKit's tautomer transforms are based on.
![2024-05-01-Tautomer-Sources-Comparison_162_0](https://github.com/rdkit/rdkit/assets/1335522/33b08190-b0a0-42fa-a508-f225d68fd422)
